src/treasury.cairo

```diff
@@ -190,6 +387,204 @@ mod Treasury {
             return status;
         }
 
+        fn get_next_pending(self: @ContractState) -> Option<Transfer> {
+            if self.transfers_count.read() == 0 {
+                return Option::None;
+            }
+            let current_timestamp = get_block_timestamp();
+            let transfers_count = self.transfers_count.read();
+            let mut i = 0;
+            let mut next_pending_transfer = Option::None;
+            while i < transfers_count {
+                let current_transfer = self.transfers_on_cooldown.read(i);
+                if current_transfer.cooldown_end > current_timestamp
+                    && current_transfer.status == TransferStatus::PENDING {
+                    next_pending_transfer = Option::Some(current_transfer);
+                    break;
+                }
+                i += 1;
+            };
+            next_pending_transfer
+        }
+
+        fn get_failed_transfers(self: @ContractState) -> Span<Transfer> {
+            match self.get_next_pending() {
+                Option::None => self
+                    .get_transfers_by_status(
+                        TransferStatus::PENDING, 0, self.transfers_count.read()
+                    ),
+                Option::Some(next_pending) => self
+                    .get_transfers_by_status(TransferStatus::PENDING, 0, next_pending.id)
+            }
+        }
+
+        fn get_live_transfers(self: @ContractState) -> Span<Transfer> {
+            let next_pending = self.get_next_pending().expect(Errors::NO_TRANSFERS);
+            self
+                .get_transfers_by_status(
+                    TransferStatus::PENDING, next_pending.id, self.transfers_count.read()
+                )
+        }
+
+        fn get_cancelled_transfers(self: @ContractState) -> Span<Transfer> {
+            self.get_transfers_by_status(TransferStatus::CANCELLED, 0, self.transfers_count.read())
+        }
+
+        fn get_finished_transfers(self: @ContractState) -> Span<Transfer> {
+            match self.get_next_pending() {
+                Option::None => self
+                    .get_transfers_by_status(
+                        TransferStatus::FINISHED, 0, self.transfers_count.read()
+                    ),
+                Option::Some(next_pending) => self
+                    .get_transfers_by_status(TransferStatus::FINISHED, 0, next_pending.id)
+            }
+        }
```

Comment:
So if there is no pending transfer, this first iterates over all transfers and then again iterates over all transfers?